algo/src/main/java/com/hartwig/serve/dao/ServeDAO.java

```diff
@@ -158,52 +146,100 @@ private static void writeActionableHotspotBatch(@NotNull Timestamp timestamp, @N
+            inserter.execute();
+        }
+    }
+
+    private static void writeActionableCodonsBatch(@NotNull Timestamp timestamp, @NotNull InsertValuesStep18 inserter,
+            @NotNull ActionableRange actionableCodon) {
+        inserter.values(timestamp,
+                actionableCodon.gene(),
+                actionableCodon.chromosome(),
+                actionableCodon.start(),
+                actionableCodon.end(),
+                actionableCodon.applicableMutationType(),
+                actionableCodon.source(),
+                actionableCodon.sourceEvent(),
+                concat(actionableCodon.sourceUrls()),
+                actionableCodon.treatment().name(),
+                concat(actionableCodon.treatment().sourceRelevantTreatmentApproaches()),
+                concat(actionableCodon.treatment().relevantTreatmentApproaches()),
+                actionableCodon.applicableCancerType().name(),
+                actionableCodon.applicableCancerType().doid(),
+                concat(toStrings(actionableCodon.blacklistCancerTypes())),
+                actionableCodon.level(),
+                actionableCodon.direction(),
+                concat(actionableCodon.evidenceUrls()));
+    }
+
+    private void writeActionableExons(@NotNull Timestamp timestamp, @NotNull List<ActionableRange> exons) {
+        for (List<ActionableRange> batch : Iterables.partition(exons, DatabaseUtil.DB_BATCH_INSERT_SIZE)) {
+            InsertValuesStep18 inserter = context.insertInto(ACTIONABLEEXON,
+                    ACTIONABLEEXON.MODIFIED,
+                    ACTIONABLEEXON.GENE,
+                    ACTIONABLEEXON.CHROMOSOME,
+                    ACTIONABLEEXON.START,
+                    ACTIONABLEEXON.END,
+                    ACTIONABLEEXON.APPLICABLEMUTATIONTYPE,
+                    ACTIONABLEEXON.SOURCE,
+                    ACTIONABLEEXON.SOURCEEVENT,
+                    ACTIONABLEEXON.SOURCEURLS,
+                    ACTIONABLEEXON.TREATMENT,
+                    ACTIONABLEEXON.SOURCETREATMENTAPPROACH,
+                    ACTIONABLEEXON.TREATMENTAPPROACH,
+                    ACTIONABLEEXON.APPLICABLECANCERTYPE,
+                    ACTIONABLEEXON.APPLICABLEDOID,
+                    ACTIONABLEEXON.BLACKLISTCANCERTYPES,
+                    ACTIONABLEEXON.LEVEL,
+                    ACTIONABLEEXON.DIRECTION,
+                    ACTIONABLEEXON.EVIDENCEURLS);
+            batch.forEach(entry -> writeActionableExonsBatch(timestamp, inserter, entry));
             inserter.execute();
         }
     }
 
-    private static void writeActionableRangeBatch(@NotNull Timestamp timestamp, @NotNull InsertValuesStep18 inserter,
-            @NotNull ActionableRange actionableRange) {
+    private static void writeActionableExonsBatch(@NotNull Timestamp timestamp, @NotNull InsertValuesStep18 inserter,
+                                                   @NotNull ActionableRange actionableExon) {
```

Comment:
Is this identical to `writeActionableCodonsBatch`? It may be redundant.

Reply:
yes, I have also merged. It was indeed duplicated